src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/programsBuilder.js

```diff
@@ -176,10 +176,16 @@ function getBuiltPrograms(
     );
 
     const promisePrograms = cachedPrograms
-        .filter(cachedProgram => (
-            cachedProgram.trackedEntityTypeId &&
-            trackedEntityTypeCollection.get(cachedProgram.trackedEntityTypeId)
-        ))
+        .filter((cachedProgram) => {
+            // allow event programs
+            if (cachedProgram.programType === 'WITHOUT_REGISTRATION') {
+                return true;
+            }
+            // We allow only tracker programs that their tracker id exists in the collection.
+            // This is because a user might have access to read and write to a program BUT
+            // they might not have access to read and write to the tracked entity type the program belongs to.
+            return (cachedProgram.trackedEntityTypeId && trackedEntityTypeCollection.get(cachedProgram.trackedEntityTypeId));
+        })
```

Comment:
Why is this change bundled with replacing travis?

Reply:
whoops that shouldnt be here. was doing some branch juggling on friday and must have missed this bit